app/assets/javascripts/discourse/controllers/add-post-notice.js.es6

```diff
@@ -0,0 +1,38 @@
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import computed from "ember-addons/ember-computed-decorators";
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  notice: null,
+  saving: false,
```

Comment:
`resolve` should be added as a data member here so that the shape of the JS object needn't change.